src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -934,9 +935,10 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
     checkIsNotNull(values);
     assertNotNull(info, actual);
     int actualSize = sizeOf(actual);
-    if (values.length != actualSize)
-      throw failures.failure(info, shouldHaveSameSize(actual, values, actualSize, values.length, comparisonStrategy));
-    assertHasSameSizeAs(info, actual, values); // include check that actual is not null
```

Comment:
I removed this, as it wasn't doing anything (the null check the comment mentioned is done a few lines up).